ipld/car/src/error.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::multihash::DecodeOwnedError;
+use std::error;
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    ParsingError(String),
+    InvalidFile(String),
+    Other(String),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Error::ParsingError(err) => write!(f, "Failed to parse CAR file: {}", err.clone()),
+            Error::InvalidFile(err) => write!(f, "Invalid CAR file: {}", err.clone()),
+            Error::Other(err) => write!(f, "Other cid Error: {}", err.clone()),
+        }
+    }
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        use self::Error::*;
+
+        match self {
+            ParsingError(_) => "Failed to parse CAR file",
+            Error::InvalidFile(_) => "Invalid CAR file",
+            Other(_) => "Other Cid Error",
```

Comment:
why the use if you are going to specify the enum prefix? Also, this is consuming the error messages, what is the point of this error type? If it doesn't matter then just remove this type and make the error type `&'static str`

Reply:
True